### ISSUE TITLE
Hero Khaz Bug / 45580:43757

### ISSUE THREAD

**Issue by cigumo** [cigumo/krli#330]
>   I was playing with hero Khaz on the "Temple of Saqra" stage, to level
>   him up. I have noticed that he freezes when the first "Poukai Riders"
>   wave comes up. After freezing he does not attack, does not move and is
>   not targeted by most enemies, except for ranged ones.
>   Upon dying, he did not revive either.
>   I guess this happens because he either tries to summon a flying enemy
>   to his position or he is trying to rush to a flying enemy's position
>   and causes an exception.
>   I tried to reproduce the issue on another stage, the "Lost Jungle", but
>   Khaz did not freeze on "poukai riders.
>   He also does not freeze on the bee waves...
>   But he consistently froze on "poukai riders" from "Temple of Saqra".
>   Device: Samsung Galaxy J5 Prime
>   OS: Android 7.0

**Comment by cigumo** [comment]
>  Hello It appears kahz has a bug where he just stands still until the end of the game it happened to me more than 5 times and I would really  appreciate it if you fix it asap.
>   It happens in any stage if I don't control him and I think it's his   charge attack that triggers the bug and yes it also happened in water   stages.
>  51707:43757

**Comment by cigumo** [comment]
I have not been able to reproduce it yet.

**Comment by sunnydaler** [comment]
https://streamable.com/9nu4w

**Comment by cigumo** [comment]
http://support.ironhidegames.com/a/tickets/58605
> Recently I bought the hero Kahz. I played through half of the game and suddenly a bug appeared on every map. After a few waves I can't interact with the hero. It can only receive damage and die (can't fight). I am running the game on Xiaomi A1, 64 GB with Android One (version 9). I assume that the bug is triggered by his "Deadalus' maze" ability, because it appeared after I upgraded it (but this might be a wrong assumption...)

**Comment by cigumo** [comment]
Still can't reproduce it.

**Comment by cigumo** [comment]
https://i.imgur.com/VqdpvAa.mp4

**Comment by cigumo** [comment]
kr2-desktop-3.2.05 : prevents Kahz from rushing into an invalid terrain

**Comment by sunnydaler** [comment]
https://streamable.com/ojdid 
https://streamable.com/0lcas
it seems that freezing happens around the same place in Temple of Saqra.
place him there in wave 11 and u might be able to reproduce it
kr2-desktop-3.2.01
```
[3134.6183] systems.ERROR    on_update() - Error running coro: ./all/utils.lua:9
41: attempt to index field 'heading' (a nil value)
stack traceback:
        ./all/utils.lua: in function 'melee_slot_position'
        ./kr2/game_scripts.lua:17042: in function 'filter_func'
        ./all/utils.lua:649: in function 'find_first_target'
        ./kr2/game_scripts.lua:17037: in function <./kr2/game_scripts.lua:16885>
```

**Comment by cigumo** [comment]
I'm pretty sure it's fixed in kr2-desktop-3.2.05 but testing will tell.
